Teensy/src/rfModule/rfModule.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file RF Module handling functions.
+ * @brief 
+ * 
+ *  functions to setup and transmit data from teensey, sets the nrf24l01 module
+ *  as a transmitter.
+ *  
+ *
+ *
+ */
+
+#include <Arduino.h>
+#include <SPI.h>
+#include "rfModule.h"
+
+#include "RF24.h"
+#include "nRF24L01.h" 
+
+#define CE_PIN 9 //Theese can be any
+#define CSN_PIN 10 
+
+// MOSI: - pin 11 
+// MISO: - pin 12
+// SCK: - pin 13
+
+RF24 radio(CE_PIN, CSN_PIN);
+const byte address[6] = "00001";
+const byte checksum = 0;
+const char newline = '\n';
+
+bool rfModule_init() 
+{
+
+    if (!radio.begin()) {
+        //may need to wait here
+        Serial.println("ERROR: Unable to connect to rfModule");
+        return false;
+    }
+
+    radio.setDataRate(RF24_250KBPS);           // greater = more distance
+    radio.setPALevel(RF24_PA_MAX);             // alternative is RF24_PA_LOW.
+    
+    radio.setPayloadSize(4);                   //set size of what is being sent (32 max) 4 or 32
+    radio.openWritingPipe(address);            //Set to TX
+    radio.stopListening();
+
+    // could send it ints at a time i.e 4 bytes at a time or as 32 byte chunks
+    
+
+
+    return true;
+}
+
+
+bool rfModule_sendline(sensorData_s *pSensorData)
+{
+    //write the data using pointers to sensor data + checksum + newline
+    radio.write(&pSensorData->wheelSpeed_mph.fr, sizeof(pSensorData->wheelSpeed_mph.fr)); 
```

Comment:
To make it easier to parse the data on the other side we want to split all the data with commas. 

I would say the best way to handle this is to create a static 32 byte char array to you can use something like snprintf to create your formatted data chunk before calling radio.write on it. I don't know exactly how .write function works but I would have thought there shouldn't be any issue with using the same array for each call

Reply:
Ok ill take a look